### PR TITLE
upgraded react-native to 0.63.2

### LINKED ISSUE #1851: java.lang.IllegalArgumentExceptionAirMapManager.java:308 - southern latitude exceeds northern latitude (NaN > NaN)
## Error in Plant-for-the-Planet App

**java.lang.IllegalArgumentException** in **MainActivity**
southern latitude exceeds northern latitude (NaN > NaN)

[View on Bugsnag](https://app.bugsnag.com/plant-for-the-planet/plant-for-the-planet-app/errors/5e2a2659ff1dd80019d14cdb?event_id=5e2a26590055e6e6f27f0000&i=gh&m=ci)

## Stacktrace

```java
java.lang.IllegalArgumentException southern latitude exceeds northern latitude (NaN > NaN) 
    Unknown:38 com.google.android.gms.common.internal.Preconditions.checkArgument
    Unknown:7 com.google.android.gms.maps.model.LatLngBounds.<init>
    AirMapManager.java:308 com.airbnb.android.react.maps.AirMapManager.receiveCommand
    AirMapManager.java:28 com.airbnb.android.react.maps.AirMapManager.receiveCommand
    NativeViewHierarchyManager.java:783 com.facebook.react.uimanager.NativeViewHierarchyManager.dispatchCommand
    UIViewOperationQueue.java:278 com.facebook.react.uimanager.UIViewOperationQueue$DispatchCommandOperation.execute
    UIViewOperationQueue.java:779 com.facebook.react.uimanager.UIViewOperationQueue$1.run
    UIViewOperationQueue.java:888 com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches
    UIViewOperationQueue.java:42 com.facebook.react.uimanager.UIViewOperationQueue.access$2200
    UIViewOperationQueue.java:948 com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded
    GuardedFrameCallback.java:28 com.facebook.react.uimanager.GuardedFrameCallback.doFrame
    ReactChoreographer.java:174 com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame
    ChoreographerCompat.java:84 com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame
    Choreographer.java:909 android.view.Choreographer$CallbackRecord.run
    Choreographer.java:723 android.view.Choreographer.doCallbacks
    Choreographer.java:655 android.view.Choreographer.doFrame
    Choreographer.java:897 android.view.Choreographer$FrameDisplayEventReceiver.run
    Handler.java:789 android.os.Handler.handleCallback
    Handler.java:98 android.os.Handler.dispatchMessage
    Looper.java:164 android.os.Looper.loop
    ActivityThread.java:6944 android.app.ActivityThread.main
    Method.java:-2 java.lang.reflect.Method.invoke
    Zygote.java:327 com.android.internal.os.Zygote$MethodAndArgsCaller.run
    ZygoteInit.java:1374 com.android.internal.os.ZygoteInit.main
```
[View full stacktrace](https://app.bugsnag.com/plant-for-the-planet/plant-for-the-planet-app/errors/5e2a2659ff1dd80019d14cdb?event_id=5e2a26590055e6e6f27f0000&i=gh&m=ci)

*Created automatically via Bugsnag*

### LINKED ISSUE #2081: Upgrade react-native to 0.63.x
A new stable version is out: https://github.com/facebook/react-native/releases/tag/v0.63.2

- [ ] Study the Changelog for deprecated features which we might need to adapt in https://github.com/react-native-community/releases/blob/master/CHANGELOG.md#v0632

- [x] Use Upgrade-Helper to check for changes in the project files https://react-native-community.github.io/upgrade-helper/?from=0.61.5&to=0.63.2

As the current react-native version is using Android SDK 28 and the new one is using Android SDK 29 and Google set a deadline for updates with SDK 28 until November, this issue also has the same deadline: https://developer.android.com/distribute/play-policies

### LINKED ISSUE #2718: com.facebook.react.bridge.JSApplicationIllegalArgumentException in MainActivity - Error while updating property 'coordinate' of a view managed by: AIRMapMarker
## Error in Plant-for-the-Planet App

**com.facebook.react.bridge.JSApplicationIllegalArgumentException** in **MainActivity**
Error while updating property 'coordinate' of a view managed by: AIRMapMarker

[View on Bugsnag](https://app.bugsnag.com/plant-for-the-planet/plant-for-the-planet-app/errors/5f5cf4aeedaedc0017bc58cc?event_id=5f5cf4ae005dfa53f35e0000&i=gh&m=ci)

## Stacktrace

    ViewManagersPropertyCache.java:102 - com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp

[View full stacktrace](https://app.bugsnag.com/plant-for-the-planet/plant-for-the-planet-app/errors/5f5cf4aeedaedc0017bc58cc?event_id=5f5cf4ae005dfa53f35e0000&i=gh&m=ci)

*Created automatically via Bugsnag*

### PR DESCRIPTION
**Pull request by norbertschuler**
Fixes #2081
Fixes #2718 and #1956
Fixes #1851

Changes in this pull request:
- upgrade packages to react-native 0.63.2
- upgraded Android project
- upgrade iOS project
- used patched versions of https://github.com/norbertschuler/react-native-material-dropdown.git and https://github.com/norbertschuler/react-native-material-textfield.git
- used patched version of https://github.com/norbertschuler/react-native-modalbox.git

Open issue:
- [ ] can not use Hermes engine for Android as @react-native-community/datetimepicker is crashing - maybe we need to replace it with another module, e.g. https://github.com/henninghall/react-native-date-picker

You best should delete the `node_modules` folder and install all modules newly and also clean you Android and iOS project to be able to build it again. Use our [clean.sh](https://github.com/Plant-for-the-Planet-org/treecounter-app/blob/develop/clean.sh) script or parts of it.

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -87,6 +89,7 @@ if (propfile.exists()) {
     properties.load(propfile.newDataInputStream())
 }
 
+// TODO: Hermes engine crashes with @react-native-community/datetimepicker
 project.ext.react = [
         entryFile   : "index.native.js",
         enableHermes: false,
```
Comment (norbertschuler) [inline android/app/build.gradle:95]: Change this to `true` to test with Herms engine enabled.

**Comment by norbertschuler** [comment]
@sagararyal @Chiagozie-Udeh maybe you can test this pull request in BrowserStack (TreecounterApp-2694.ipa and TreecounterApp-2694.apk)? I already tested, but probably not every function of the app - until now it works for me.) I think this upgrade is necessary if we ever want to continue upgrading npm modules for this app (see Snyk pull requests) to avoid security problems.

**Comment by Chiagozie-Udeh** [comment]
> @sagararyal @Chiagozie-Udeh maybe you can test this pull request in BrowserStack (TreecounterApp-2694.ipa and TreecounterApp-2694.apk)? I already tested, but probably not every function of the app - until now it works for me.) I think this upgrade is necessary if we ever want to continue upgrading npm modules for this app (see Snyk pull requests) to avoid security problems.

On it now @norbertschuler

**Comment by Chiagozie-Udeh** [comment]
iOS Upgrade:

- Country doesn't work. It is clickable but doesn't show (both in donations and gift trees for all purposes)
- Others work fine
Capture1

Android upgrade:
- Country doesn't work. It is clickable but doesn't show (both in donations and gift trees for all purposes)
- Register tree returns error and exits the app or just exits the app abruptly without completing the register tree task. This is for both single and multiple registration with and without photos 
- Others work fine 
Capture-

**Comment by norbertschuler** [comment]
@Chiagozie-Udeh thank you for testing. I replaced the old (and since three years unsupported `react-native-material-dropdown` module with the still supported `@react-native-community/picker`).

**Comment by norbertschuler** [comment]
> * Register tree returns error and exits the app or just exits the app abruptly without completing the register tree task. This is for both single and multiple registration with and without photos

@Chiagozie-Udeh I cannot reproduce this error. Which device have you tested with?

**Comment by norbertschuler** [comment]
> > * Register tree returns error and exits the app or just exits the app abruptly without completing the register tree task. This is for both single and multiple registration with and without photos
> 
> @Chiagozie-Udeh I cannot reproduce this error. Which device have you tested with?

I found out this crash already existed for a longer time and is not newly introduced by this pull request: https://github.com/Plant-for-the-Planet-org/treecounter-app/issues?q=is%3Aissue+is%3Aopen+com.facebook.react.bridge.JSApplicationIllegalArgumentException

**Pull request merged**